scripts/tests/test_zkai_d64_gate_value_projection_proof_input.py

```
@@ -0,0 +1,100 @@
+            GATE_VALUE.validate_payload(payload)
+
+    def test_payload_rejects_projection_row_commitment_drift(self) -> None:
+        payload = self.fresh_payload()
+        payload["gate_value_projection_mul_row_commitment"] = "blake2b-256:" + "55" * 32
+        with self.assertRaisesRegex(GATE_VALUE.GateValueProjectionInputError, "gate/value projection row"):
+            GATE_VALUE.validate_payload(payload)
+
+    def test_payload_rejects_gate_matrix_root_drift(self) -> None:
+        payload = self.fresh_payload()
+        payload["gate_matrix_root"] = "blake2b-256:" + "77" * 32
+        with self.assertRaisesRegex(GATE_VALUE.GateValueProjectionInputError, "gate_matrix_root"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)